crates/fuel-core/src/database/state.rs

```diff
@@ -231,29 +232,23 @@ mod tests {
             }
 
             // then
-            let states_in_db: Vec<_> = database
+            let mut states_in_db: Vec<_> = database
                 .iter_all::<ContractsState>(None)
                 .collect::<Result<Vec<_>, _>>()
                 .unwrap()
                 .into_iter()
-                .map(|(key, value)| {
-                    let contract_id = *key.contract_id();
-                    let key = *key.state_key();
-
-                    ContractStateConfig {
-                        contract_id,
-                        key,
-                        value: value.0,
-                    }
-                })
+                .map(|(_key, value)| value)
                 .collect();
 
-            let original_state = state_groups
+            let mut original_state = state_groups
                 .into_iter()
                 .flatten()
-                .sorted()
+                .map(|v| v.value)
                 .collect::<Vec<_>>();
 
+            states_in_db.sort_by(|a, b| a.as_ref().cmp(b.as_ref()));
+            original_state.sort_by(|a, b| a.as_ref().cmp(b.as_ref()));
```

Comment:
Before we were comparing `ContractStateConfig`, now we only compare values. Why do we want to change it?

Reply:
751c407b0 Now back to comparing both keys and values